### ISSUE TITLE
Jobs hung in READY

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#601]
In GitLab by @mcastril on Oct 15, 2020, 13:44

Hi @dbeltrankyl . @ mfalls has this workflow with a wrapper in a34f 

http://bscesweb04.bsc.es/autosubmitapp/experiment/a34f

It seems there was a problem when some of the inner jobs failed.

```
2020-10-13 19:04:59,390 Command sacct -n -j 11971941 -o JobId%25,State,NCPUS,NNodes,Submit,Start,End,ConsumedEnergy,MaxRSS%25,AveRSS%25 in mn-bsc32 successful with out message:                  11971941     FAILED          2        1 2020-10-13T19:01:37 2020-10-13T19:03:09 2020-10-13T19:03:21          1.76K
           11971941.batch     FAILED          2        1 2020-10-13T19:03:09 2020-10-13T19:03:09 2020-10-13T19:03:21          1.35K                      815K                      815K
          11971941.extern  COMPLETED          2        1 2020-10-13T19:03:09 2020-10-13T19:03:09 2020-10-13T19:03:22          1.76K                      538K                      538K

2020-10-13 19:04:59,390 Output                  11971941     FAILED          2        1 2020-10-13T19:01:37 2020-10-13T19:03:09 2020-10-13T19:03:21          1.76K                                                  
           11971941.batch     FAILED          2        1 2020-10-13T19:03:09 2020-10-13T19:03:09 2020-10-13T19:03:21          1.35K                      815K                      815K
          11971941.extern  COMPLETED          2        1 2020-10-13T19:03:09 2020-10-13T19:03:09 2020-10-13T19:03:22          1.76K                      538K                      538K

2020-10-13 19:04:59,413 Updating FAILED jobs
2020-10-13 19:04:59,413 Resetting job: a34f_19600101_fc0000_1_SIM status to: READY for retrial...
2020-10-13 19:04:59,413 Resetting job: a34f_19600101_fc0000_2_SIM status to: READY for retrial...
2020-10-13 19:04:59,413 Update finished
2020-10-13 19:04:59,413 Updating WAITING jobs
2020-10-13 19:04:59,413 Update finished
2020-10-13 19:04:59,413
Job Name                           Job Id         Job Status     Job Platform        Job Queue
2020-10-13 19:04:59,414 a34f_INI                           11971927       COMPLETED      marenostrum4        debug
2020-10-13 19:04:59,417 Saving JobList: /esarchive/autosubmit/a34f/pkl/job_list_a34f.pkl
2020-10-13 19:04:59,418 Job list saved
2020-10-13 19:04:59,420
Jobs ready for marenostrum4: 1
2020-10-13 19:04:59,420 Number of jobs available: 6
2020-10-13 19:04:59,421 Number of jobs ready: 1
2020-10-13 19:04:59,421 Jobs ready for marenostrum4: 1
2020-10-13 19:04:59,431 Custom directives from platform.conf: None
2020-10-13 19:04:59,655 a34f_19600101_fc0000_2_SIM_STAT have been removed
2020-10-13 19:05:00,805 Command /gpfs/scratch/bsc32/bsc32786/a34f/LOG_a34f/submit_marenostrum4.sh in mn-bsc32 successful with out message: Submitted batch job 11972003

2020-10-13 19:05:00,805 Output Submitted batch job 11972003
```

As you can see, jobs were moved to READY, but after that it seems it is not finding any ready job:

```
6 of 7 jobs remaining (19:05)
2020-10-13 19:05:20,880 Sleep: 10
2020-10-13 19:05:20,880 Number of retrials: 3
2020-10-13 19:05:20,881 WRAPPER CHECK TIME = 10
2020-10-13 19:05:20,881 Updating FAILED jobs
2020-10-13 19:05:20,881 Update finished
2020-10-13 19:05:20,881 Updating WAITING jobs
2020-10-13 19:05:20,882 Update finished
2020-10-13 19:05:20,882
Jobs ready for marenostrum4: 0
2020-10-13 19:05:20,882 Number of jobs available: 6
2020-10-13 19:05:20,882 Number of jobs ready: 0
2020-10-13 19:05:20,885 Updating FAILED jobs
2020-10-13 19:05:20,885 Update finished
2020-10-13 19:05:20,885 Updating WAITING jobs
2020-10-13 19:05:20,885 Update finished
2020-10-13 19:05:30,898 Reloading parameters...
2020-10-13 19:05:30,912 Loading parameters...
2020-10-13 19:05:30,913 Loading project parameters...
2020-10-13 19:05:30,914

```

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 15, 2020, 13:46

@ mfalls , you can maybe do a backup of `/esarchive/autosubmit/a34f/pkl` so @dbeltrankyl can take a look to that, and then start from scratch because it seems Autosubmit failed when one of those SIM failed.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Oct 15, 2020, 13:46

Hi, 

Is this on 3.13.0b? I believe that is related to #599 and a fix is incoming

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 15, 2020, 13:51

yes it is

**Comment by kinow** [comment]
In GitLab by @ lpalma on Oct 16, 2020, 14:17

Ok, thanks for the clarification @dbeltrankyl!!

**Comment by kinow** [comment]
In GitLab by @etiennesky on Oct 18, 2020, 07:36

Hi @dbeltrankyl can you please consider fixing this quite serious issue in 3.12.0 ? 

You say bugfixes should go in 3.13.0 but since you cannot guarantee is doesn't have any bugs, that is not an option for running production experiments as the ones in https://earth.bsc.es/gitlab/es/auto-ecearth3/-/issues/1427

This bug is a showstopper for me since I have to kill AS more than twice a day to get my experiments running.

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 18, 2020, 17:40

@etiennesky the connection issue is mostly affecting CCA experiments, and this is because there is an ongoing issue in CCA connectivity (we have been testing 3.12.0 in CCA without issues till now and it passed the "test period" that took several months).

I think it would be less dangerous that you guys use 3.13.0 while ecaccess is being fixed, that changing 3.12.0 that is used also for MN4 workflows and hasn't been touched in months. That's for example what the AC guys have been doing with Data assimilation experiments (because they have specific workflow necessities, not relevant for the rest).

This changes are not trivial, more than a bugfix we could say that it's part of the major rework to make Autosubmit more stable.

**Comment by kinow** [comment]
In GitLab by @etiennesky on Oct 20, 2020, 19:58

mentioned in issue auto-ecearth3#1427

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Oct 20, 2020, 21:13

hi @etiennesky ,

use 3.12.0-fixed there is an issue with a library(portalocker) on 3.12.0 working on it but 3.12.0-fixed should work now

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 21, 2020, 10:27

Thanks @dbeltrankyl.

> I am having this issue now with AS 3.12.0 (the default, I did not change any modules) with expid a36z

@etiennesky so in the end what it happened is that as you were guessing, 3.12.0-fixed and 3.12.0 were pointing to the new one. But the error that you reported was due to the loss of the connection, that in the patched version should be managed with the setting of the job to UNKNOWN. There was a bug in this new part that was fixed. The "positive" side, as @dbeltrankyl mentioned, is that instead of that crash, your Autosubmit would have entered in the "session not active" infinite loop, so it wouldn't have been any better.

This particular issue is very difficult to reproduce because it depends on the environment, and I don't know if this would help in this particular case, but I will open an issue that I had in the backlog to create a small testing suite for Autosubmit, with a small experiment from every Auto-model.

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 27, 2020, 12:01

closed
